clippy_lints/src/loops/never_loop.rs

```diff
@@ -234,16 +241,19 @@ fn never_loop_expr_all<'a, T: Iterator<Item = &'a Expr<'a>>>(
     main_loop_id: HirId,
 ) -> NeverLoopResult {
     es.map(|e| never_loop_expr(e, ignore_ids, main_loop_id))
-        .fold(NeverLoopResult::Otherwise, combine_both)
+        .fold(NeverLoopResult::Otherwise, combine_seq)
 }
 
 fn never_loop_expr_branch<'a, T: Iterator<Item = &'a Expr<'a>>>(
     e: &mut T,
     ignore_ids: &mut Vec<HirId>,
     main_loop_id: HirId,
 ) -> NeverLoopResult {
+    let ignore_ids_saved = ignore_ids.clone();
```

Comment:
You shouldn't need to clone here. By the time `never_loop_expr` returns either `ignore_ids` is back to it's original state, or you have a bug which needs to be fixed.

Reply:
The `.fold()` executes at the same time as the iterator, not at the end, I cannot borrow `ignore_ids` both in the `.map()` (mutably) and in the `.fold()`. So either I clone `ignore_ids`, or I `.collect()` the results of the map.

Reply:
You can inline the `map` into the `fold`.

Reply:
Done.